src/Chapter3.hs

```diff
@@ -971,6 +1071,27 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data WeekDay = Monday
+              | Tuesday
+              | Wednesday
+              | Thursday
+              | Friday
+              | Saturday
+              | Sunday
+              deriving (Show, Eq, Ord, Enum)
+
+isWeekend :: WeekDay -> Bool
+isWeekend Saturday = True
+isWeekend Sunday = True
+isWeekend _ = False
+
+nextDay :: WeekDay -> WeekDay
+nextDay Sunday = Monday --Not sure if enum will handle this already, including just to be safe
+nextDay day = succ day
```

Comment:
Yes, you are right that `Enum` won't handle this. By default, `succ` fails in runtime on the last element of enumeration. But here you want a cyclic note.

For example, we've implemented such polymorphic function in our alternative standard library:

* http://hackage.haskell.org/package/relude-0.7.0.0/docs/Relude-Extra-Enum.html#v:next